core/model/collection.py

```diff
@@ -51,34 +48,38 @@ class Collection(Base, HasSessionCache):
     __tablename__ = "collections"
     id = Column(Integer, primary_key=True, nullable=False)
 
-    name = Column(Unicode, unique=True, nullable=False, index=True)
+    # TODO: This should no longer be used. And will be removed in the next release.
+    #  Collections store their configurations in integration configurations now.
+    #  This is only left here in case there needs to be a rollback to the current
+    #  release.
+    _name_deprecated = Column("name", Unicode)
 
     DATA_SOURCE_NAME_SETTING = "data_source"
 
-    # For use in forms that edit Collections.
-    EXTERNAL_ACCOUNT_ID_KEY = "external_account_id"
+    # TODO: This should no longer be used. And will be removed in the next release.
+    #  Collections store their configurations in integration configurations now.
+    #  This is only left here in case there needs to be a rollback to the current
+    #  release.
+    _external_account_id_deprecated = Column("external_account_id", Unicode)
 
-    # How does the provider of this collection distinguish it from
-    # other collections it provides? On the other side this is usually
-    # called a "library ID".
-    external_account_id = Column(Unicode, nullable=True)
+    # TODO: This should no longer be used. And will be removed in the next release.
+    #  Collections store their configurations in integration configurations now.
+    #  This is only left here in case there needs to be a rollback to the current
+    #  release.
+    _external_integration_id_deprecated = Column("external_integration_id", Integer)
```

Comment:
This keeps the column, but removes the constraints on it. This means we should be able to roll back if we need to, but otherwise this column should be removed soon after as it will begin to get stale.